cli/src/pcluster/validators/ec2_validators.py

```diff
@@ -171,10 +171,15 @@ class KeyPairValidator(Validator):
     Verify the given key pair is correct.
     """
 
-    def _validate(self, key_name: str):
+    def _validate(self, key_name: str, os: str):
         if key_name:
             try:
-                AWSApi.instance().ec2.describe_key_pair(key_name)
+                key_pair_data = AWSApi.instance().ec2.describe_key_pair(key_name)
+                if os == "ubuntu2204" and key_pair_data["KeyPairs"][0]["KeyType"] == "rsa":
+                    self._add_failure(
+                        "Ubuntu 22.04 does not support RSA keys.  Please generate and use an ed25519 key",
```

Comment:
you can think to generalize this logic, in future AL2023 will have to do the same check, so we could create a dictionary and do something like:

```python
os_key_type = {"ubuntu22": "ed25519"}
if os in os_key_type:
    required_key_type = os_key_type.get(os)
    if key_pair.key_type != required_key_type:
        self._add_failure(f"{os} requires {required_key_type} key. Please generate and use a {required_key_type} key",...
```

Reply:
All the keys we generate are ED25519 now, but that doesn't stop users from using different key types like ECDSA.  I think we should avoid this complexity for now and we can add the dict when we have more than one case to consider